ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -814,7 +841,31 @@ func (r *RayClusterReconciler) SetupWithManager(mgr ctrl.Manager, reconcileConcu
 			predicate.LabelChangedPredicate{},
 			predicate.AnnotationChangedPredicate{},
 		))).
-		Watches(&source.Kind{Type: &corev1.Event{}}, &handler.EnqueueRequestForObject{}).
+		Watches(&source.Kind{Type: &corev1.Event{}},
+			&handler.EnqueueRequestForObject{},
+			builder.WithPredicates(predicate.Funcs{
+				CreateFunc: func(e event.CreateEvent) bool {
+					if eventObj, ok := e.Object.(*corev1.Event); ok {
+						if eventObj.InvolvedObject.Kind != "Pod" || eventObj.Type != "Warning" ||
+							eventObj.Reason != "Unhealthy" || !strings.Contains(eventObj.Message, "Readiness probe failed") {
```

Comment:
Is there a more programmatic way to get the cause of the unhealthiness, or is scanning the message string standard practice?  Also, for my education, why do we only care about failures in the readiness probe and not other pod unhealthy events?  Maybe @kevin85421 you can also respond if you know the answer

Reply:
readiness probe failure will trigger liveness probe failure. readiness probe will not automatically trigger a pod rebuild but liveness probe failure will.